client/consensus/slots/src/slots.rs

```diff
@@ -71,6 +71,8 @@ pub fn time_until_next(now: Duration, slot_duration: u64) -> Duration {
 pub struct SlotInfo {
 	/// The slot number.
 	pub number: u64,
+	/// The last slot number produced.
+	pub last_number: u64,
```

Comment:
This parameter not currently used, but is added so that engines can implement logic on `proposing_remaining_duration` based on parent/last proposing slot.

Reply:
What does `produced` mean here?

What we would probably want to compare against is the slot of the parent header

Reply:
@rphmeier It should be the slot when we last tried to propose blocks, so it's not necessarily `number - 1`. I think this parameter would be useful if we just want to calculate the `remaining_duration` based on past rough average proposing time?

I added reference of chain head header to `proposing_remaining_duration`. In this way, if the engine wants to calculate the duration based on parent slot, they can do it by parsing the header digest.

Reply:
I think that's the best we could hope for. Thanks